pints/_core.py

```diff
@@ -341,3 +341,28 @@ def values(self):
         points and ``n_outputs`` is the number of outputs.
         """
         return self._values
+
+
+class TunableMethod(object):
+
+    """
+    Defines an interface for a numerical method with a given number of
+    hyper-parameters.
+    """
+
+    def n_hyper_parameters(self):
+        """
+        Returns the number of hyper-parameters for this method.
```

Comment:
Bit more explanation is needed! This will appear in the docs of lots of methods, outside the context of tuning. So every docstring should be self-explanatory (and/or refer to this class)

Reply:
how is this @MichaelClerx ?

Reply:
Beautiful!